### DISCUSSION TITLE
`quarto render` quarto book to pdf

### DISCUSSION THREAD

**Discussion by xx02al** [posit-dev/positron#5539] [Q&A]
I'm still on my way to become better more familiar with positron. Recently I realized that I end up with an error when I do `quarto render` (which includes pdf) on a quarto book project in the terminal. Rendering to html works fine, in RStudio both html and pdf work fine. Kind or not, before I invest more time, I wanted to ask if there is already a clear error on my side based on this short description and/or if others observe something similar? How does it work correctly?

**Comment by juliasilge** [comment]
What happens if you do `quarto install tinytex` as outlined here? 
https://quarto.org/docs/output-formats/pdf-basics.html#prerequisites
Does that solve the problem?

Seems likely that RStudio is bundling some prerequisites for PDF building that Positron is not yet.

**Reply by xx02al** [reply]
It asked me whether I want to update my current installation, what I agreed on. As far as I can judge this then went well. I still end in the same scenario. Does this help figure out what I am doing wrong: ERROR: NotFound ... "C\Program Files\Positron\resources\app\quarto\share\formats\pdf\pandoc\before-bib.tex"!? As I saw then, the same error appears when rendering a "simple" quarto doc to pdf. This might be more helpful, as TinyTeX is obviously not detected globally in positron.
Honestly, I don't really understand what is meant by `quarto install tinytex --update-path` in the documentation - can it be that quarto can recognize TinyTeX in RStudio, but not in positron (even if TinyTeX was updated from there)? Should I try `quarto install tinytex --update-path` or am I doing something wrong?

**Reply by juliasilge** [reply]
I don't think you're doing anything wrong @xx02al, no. I think what is happening is that Positron is bundling Quarto but not a PDF engine such as tinytex.

What _we_ need to do is bundle tinytex or similar along with Quarto, similarly to how RStudio bundles a PDF engine.

What _you_ need to do as a workaround for now to get unstuck is to get a version of Quarto installed that knows about a PDF engine such as tinytex. I think what I would do in your situation is close Positron, [install Quarto totally fresh](https://quarto.org/docs/get-started/), do `quarto install tinytex`, and then try again with your simple Quarto doc to see if it is all set up correctly. Please let us know if you can't get unstuck!

**Reply by juliasilge** [reply]
I opened https://github.com/posit-dev/positron/issues/5553 to track the work we need to do to get this working. Thank you for the report! 🙌

**Reply by xx02al** [reply]
Thank you @juliasilge. Interesting. I have now installed the latest version of quarto (1.6.39; which I was going to do these days anyway) and with this one rendering to pdf (that is not built from typst) works, both "simple" quarto docs and books. Before updating, I had positron 2024.12.0-41 and the quarto version must have been 1.5.55 or 1.5.57.
I then tried things out on a second device, also here up until then, rendering to pdf worked in RStudio and the error described appeared in positron. I can confirm also `quarto install tinytex` with a subsequent positron restart is required after updating to quarto 1.6.39.
I think it makes sense you have opened #5553.

**Comment by cderv** [comment]
To add more context

````
ERROR: NotFound ... "C\Program Files\Positron\resources\app\quarto\share\formats\pdf\pandoc\before-bib.tex"!?
````

This is a problem in the bundled version of Quarto by Positron. There are three empty file that should be in `share\formats\pdf\pandoc\` and there are not inside the Quarto version bundled in Positron

`````
❯ ls C:\Users\chris\scoop\apps\positron-daily\current\resources\app\quarto\share\formats\pdf\pandoc\

    Directory: C:\Users\chris\scoop\apps\positron-daily\current\resources\app\quarto\share\formats\pdf\pandoc

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---          03/10/2025    07:40            368 after-header-includes.latex
-a---          03/10/2025    07:40            787 babel-lang.tex
-a---          03/10/2025    07:40            140 before-body.tex
-a---          03/10/2025    07:40            635 biblio-config.tex
-a---          03/10/2025    07:40            338 biblio.tex
-a---          03/10/2025    07:40           1265 citations.tex
-a---          03/10/2025    07:40           1489 common.latex
-a---          03/10/2025    07:40            261 doc-class.tex
-a---          03/10/2025    07:40           3481 font-settings.latex
-a---          03/10/2025    07:40            687 fonts.latex
-a---          03/10/2025    07:40            583 graphics.tex
-a---          03/10/2025    07:40            653 hypersetup.latex
-a---          03/10/2025    07:40           6482 latex.common
-a---          03/10/2025    07:40           2317 latex.template
-a---          03/10/2025    07:40           1595 pandoc.tex
-a---          03/10/2025    07:40            312 passoptions.latex
-a---          03/10/2025    07:40            653 tables.tex
-a---          03/10/2025    07:40           1083 template.tex
-a---          03/10/2025    07:40             83 tightlist.tex
-a---          03/10/2025    07:40            397 title.tex
-a---          03/10/2025    07:40            289 toc.tex
`````

compared to 
````
❯ ls C:\Users\chris\scoop\apps\quarto\current\share\formats\pdf\pandoc\

    Directory: C:\Users\chris\scoop\apps\quarto\current\share\formats\pdf\pandoc

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-----          29/09/2025    23:14              0 after-body.tex
-----          29/09/2025    23:14            368 after-header-includes.latex
-----          29/09/2025    23:14            787 babel-lang.tex
-----          29/09/2025    23:14              0 before-bib.tex
-----          29/09/2025    23:14            140 before-body.tex
-----          29/09/2025    23:14              0 before-title.tex
-----          29/09/2025    23:14            635 biblio-config.tex
-----          29/09/2025    23:14            338 biblio.tex
-----          29/09/2025    23:14           1265 citations.tex
-----          29/09/2025    23:14           1489 common.latex
-----          29/09/2025    23:14            261 doc-class.tex
-----          29/09/2025    23:14           3481 font-settings.latex
-----          29/09/2025    23:14            687 fonts.latex
-----          29/09/2025    23:14            583 graphics.tex
-----          29/09/2025    23:14            653 hypersetup.latex
-----          29/09/2025    23:14           6482 latex.common
-----          29/09/2025    23:14           2317 latex.template
-----          29/09/2025    23:14           1595 pandoc.tex
-----          29/09/2025    23:14            312 passoptions.latex
-----          29/09/2025    23:14            653 tables.tex
-----          29/09/2025    23:14           1083 template.tex
-----          29/09/2025    23:14             83 tightlist.tex
-----          29/09/2025    23:14            397 title.tex
````

See the 3 files with size 0.  Those are empty template that are expected to be present. 

So I would say there is a problem in the bundling process for Positron. 

@xx02al if this is the only error you got, and you already had an install of Tex Live (or TinyTex), as I assumed to do as `quarto install tinytex` advised you to update, then you did nothing wrong. This should be working but the Quarto version of Positron has those missing files that are required. 

You need to use a different quarto installation for now, as you did already 
> Interesting. I have now installed the latest version of quarto (1.6.39; which I was going to do these days anyway) and with this one rendering to pdf (that is not built from typst) works, both "simple" quarto docs and books. Before updating, I had positron 2024.12.0-41 and the quarto version must have been 1.5.55 or 1.5.57.
